plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/BaseIcebergConnectorSmokeTest.java

```diff
@@ -0,0 +1,121 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.iceberg;
+
+import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.ImmutableSet;
+import io.trino.testing.BaseConnectorSmokeTest;
+import io.trino.testing.QueryRunner;
+import io.trino.testing.TestingConnectorBehavior;
+import io.trino.tpch.TpchTable;
+import org.apache.iceberg.FileFormat;
+import org.testng.annotations.Test;
+
+import java.util.Map;
+import java.util.Optional;
+
+import static io.trino.plugin.iceberg.IcebergQueryRunner.createIcebergQueryRunner;
+import static java.lang.String.format;
+import static java.util.Objects.requireNonNull;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+
+public abstract class BaseIcebergConnectorSmokeTest
+        extends BaseConnectorSmokeTest
+{
+    protected final String schemaName;
+    protected final FileFormat format;
+
+    public BaseIcebergConnectorSmokeTest(String schemaName, FileFormat format)
+    {
+        this.schemaName = requireNonNull(schemaName, "schemaName is null");
+        this.format = requireNonNull(format, "format is null");
+    }
+
+    @Override
+    protected QueryRunner createQueryRunner()
+            throws Exception
+    {
+        return createQueryRunner(ImmutableMap.of());
+    }
+
+    protected QueryRunner createQueryRunner(Map<String, String> connectorProperties)
+            throws Exception
+    {
+        return createIcebergQueryRunner(
+                ImmutableMap.of(),
+                ImmutableMap.<String, String>builder()
+                        .put("iceberg.file-format", format.name())
+                        .putAll(connectorProperties)
+                        .buildOrThrow(),
+                createSchemaInitializer(schemaName).build(),
+                Optional.empty());
+    }
+
+    protected SchemaInitializer.Builder createSchemaInitializer(String schemaName)
+    {
+        return SchemaInitializer.builder()
+                .withSchemaName(schemaName)
+                .withClonedTpchTables(
+                        ImmutableSet.<TpchTable<?>>builder()
+                                .add(TpchTable.NATION, TpchTable.REGION)
```

Comment:
use REQUIRED_TPCH_TABLES

however, you don't need this method here. you can let the subclasses do the job.

Reply:
Ok so reverting to where it was at the beginning :)